src/components/AddSongMenu.tsx

```diff
@@ -0,0 +1,226 @@
+import Button from "@mui/material/Button";
+import List from "@mui/material/List";
+import ListItem from "@mui/material/ListItem";
+import ListItemButton from "@mui/material/ListItemButton";
+import ListItemText from "@mui/material/ListItemText";
+import {
+  Avatar,
+  CircularProgress,
+  InputBase,
+  ListItemAvatar,
+  Paper,
+  Typography,
+} from "@mui/material";
+import MoreVertIcon from "@mui/icons-material/MoreVert";
+import { useContext, useState, FormEvent, ChangeEvent } from "react";
+import { AppContext } from "../App";
+import toast from "react-hot-toast";
+import CancelIcon from "@mui/icons-material/Cancel";
+import { handleExpiredToken } from "../helpers/handleExpiredToken";
+
+type AddMenuProps = {
+  currentSong: string;
+};
+
+interface PlayListWithTracksInterface
+  extends SpotifyApi.PlaylistObjectSimplified {
+  songsOnPlayList: SpotifyApi.PlaylistTrackObject[];
+}
+
+export default function AddSongMenu({ currentSong }: AddMenuProps) {
+  const { spotifyApi } = useContext(AppContext);
+  const [playListWithTracks, setPlayListWithTracks] = useState<
+    PlayListWithTracksInterface[]
+  >([]);
+  const [state, setState] = useState({
+    bottom: false,
+  });
+  const [playListName, setPlayListName] = useState("");
+  const [isLoading, setIsLoading] = useState(false);
+  const showFetchingDataMessage = <CircularProgress />;
+
+  async function fetchPlayList() {
+    try {
+      setIsLoading(true);
+      const playListData = await spotifyApi.getUserPlaylists();
+      const playlistAndSongs = await Promise.all(
+        playListData.items.map(async (list) => {
+          const { items } = await spotifyApi.getPlaylistTracks(list.id);
+          const songsOnPlayList = items;
+          return { ...list, songsOnPlayList };
+        })
+      );
+
+      setPlayListWithTracks(playlistAndSongs);
+      setIsLoading(false);
+    } catch (error) {
+      handleExpiredToken(error);
+    }
+  }
+
+  const toggleDrawer = (anchor: string, open: boolean) => () => {
+    setState({ ...state, [anchor]: open });
+    if (open) {
+      fetchPlayList();
+    }
+  };
+
+  const list = (anchor: string) => {
+    return (
+      <Box
+        sx={{ width: anchor === "bottom" ? "auto" : 250 }}
+        role="presentation"
+      >
+        <List>
+          {playListWithTracks.map((list) => {
+            const included = list.songsOnPlayList.some(
+              (list) => list.track.uri === currentSong
+            );
+
+            return (
+              <ListItem
+                key={list.id}
+                disablePadding
+                onClick={() => handleListItemClick(list)}
+                onKeyDown={(event) => {
+                  if (event.key === "Enter") {
+                    handleListItemClick(list);
+                  }
+                }}
+              >
+                <ListItemButton>
+                  <ListItemAvatar>
+                    <Avatar
+                      alt={`{list.name}`}
```

Comment:
This looks weird. Shouldn't it be `alt={list.name}`?

Reply:
Yes, it should be alt={list.name}. Not sure why I did it like this alt={`{list.name}`}